### ISSUE TITLE
Cursor flickers when tablet + mouse are used at the same time

### ISSUE THREAD

**Issue by Jaymusta** [McKay42/McOsu#164]
When playing with my wacom cth-480 my cursor flickers, which is most noticeable on triples and some sliders. It causes me to miss notes and sliderbreak when I would otherwise hit notes and sliders correctly. The problem persists even when toggling raw input and tablet workaround settings. A friend of mine plays with a wacom ctl-490 and also experiences these issues.

**Comment by Jaymusta** [comment]
This twitch clip shows it happening. It's a lot easier to notice it in real-time vs playback, but just play the video at 0.25x speed on the triple about 9 seconds into the clip

**Comment by McKay42** [comment]
Cursor flickering can be caused by a non-absolute cursor position update, e.g. if you move your mouse while also hovering the tablet pen, this causes the cursor to teleport to the bottom right edge with the current logic (since it thinks you are switching between two input devices at the same time).

- Have you tried disabling Windows Ink?
- Did enabling OS TabletPC Support also not fix it, since you mentioned that you tried all Tablet workaround settings?
- Do you have any gesture recognition stuff still turned on, in Windows or in the Wacom settings?
- Have you tried [FixMyPen](https://github.com/saveenr/Fix_My_Pen/releases)?
- Are you wiggling the table with very hard keyboard tapping or your tablet pen hand, and maybe moving your mouse slightly with it? If so, try temporarily unplugging your mouse to confirm.

(I've had it on my list to rework the input logic to smoothly switch between absolute/non-absolute input like peppy osu does it, but never got around to it yet unfortunately, maybe in the coming winter holidays. My current code for it is a bit of a clusterfuck.)

**Comment by Jaymusta** [comment]
Windows Ink and gesture recognition wasn't on and fixmypen didn't solve it either.
It seems that turning on OS Tablet PC Support just made it more apparent that the mouse was moving as I tapped on the keyboard. The keyboard and mouse are on a shaky drawer, so it makes sense that the mouse moving would be the source of the problem.
I disconnected my mouse and the cursor stopped teleporting which does confirm that.
